api/src/Microsoft.Azure.IIoT.Api/src/Registry/Extensions/ModelExtensions.cs

```diff
@@ -410,6 +410,7 @@ public static class ModelExtensions {
             return new DiscovererApiModel {
                 Id = model.Id,
                 SiteId = model.SiteId,
+                SiteOrGatewayId = model.SiteOrGatewayId,
```

Comment:
Please change the SiteOrGatewayId to "GatewayId" in the service model.   Update the ToServiceModel() extension method also, to set Gateway id from api model (for testing and adapter support).   

Then make sure that the GatewayId is set as part of the "registration" extension (trace where SiteOrGatewayId is set).   instead of reading it from the registration, use the "deviceid" property, not "SiteOrGatewayId" property.   The latter is an "or", but the gatewayId is always the "deviceid".